meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper

```diff
@@ -23,40 +23,48 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
+# must exit on error
+set -e
+set -o pipefail
+
 # we must be root
 [ $(whoami) = "root" ] || { echo "E: You must be root" && exit 1; }
 
 # we must have few tools
 SGDISK=$(which sgdisk) || { echo "E: You must have sgdisk" && exit 1; }
-PARTED=$(which parted) || { echo "E: You must have parted" && exit 1; }
-PARTPROBE=$(which partprobe) || { echo "E: You must have partprobe" && exit 1; }
+FDISK=$(which fdisk) || { echo "E: You must have fdisk" && exit 1; }
+RESIZEPART=$(which resizepart) || { echo "E: You must have resizepart" && exit 1; }
 RESIZE2FS=$(which resize2fs) || { echo "E: You must have resize2fs" && exit 1; }
 
 # find root device
 ROOT_DEVICE=$(findmnt --noheadings --output=SOURCE / | cut -d'[' -f1)
 # prune root device (for example UUID)
 ROOT_DEVICE=$(realpath ${ROOT_DEVICE})
 # get the partition number and type
-PART_ENTRY_NUMBER=$(udevadm info --query=property --name=${ROOT_DEVICE} | grep '^ID_PART_ENTRY_NUMBER=' | cut -d'=' -f2)
+INFO=$(udevadm info --query=property --name=${ROOT_DEVICE})
+PART_ENTRY_NUMBER=$(echo "${INFO}" | grep '^ID_PART_ENTRY_NUMBER=' | cut -d'=' -f2)
 
 # in case the root device is not on a partitioned media
 if [ "x$PART_ENTRY_NUMBER" = "x" ]; then
 	${RESIZE2FS} "${ROOT_DEVICE}"
 	exit 0
 fi
 
-PART_TABLE_TYPE=$(udevadm info --query=property --name=${ROOT_DEVICE} | grep '^ID_PART_TABLE_TYPE=' | cut -d'=' -f2)
+PART_TABLE_TYPE=$(echo "${INFO}" | grep '^ID_PART_TABLE_TYPE=' | cut -d'=' -f2)
 # find the block device
 DEVICE=$(udevadm info --query=path --name=${ROOT_DEVICE} | awk -F'/' '{print $(NF-1)}')
 DEVICE="/dev/${DEVICE}"
 
+SIZE=$(blockdev --getsz ${DEVICE})
 if [ "$PART_TABLE_TYPE" = "gpt" ]; then
 	${SGDISK} -e ${DEVICE}
-	${PARTPROBE}
```

Comment:
At this point the backup GPT data structure will be at the end of the disk, which I believe will change the partition table as a consequence, and why we had partprobe being called here before.

I think it is better to have partprobe still being called, and for that we can enable CONFIG_PARTPROBE in busybox and use the tool from it instead.

Same after resizing the partition.